pupil-spa/src/app/services/storage/storage.service.ts

```diff
@@ -54,4 +54,23 @@ export class StorageService {
       return obj;
     }, {});
   }
+
+  mergeItems(key: StorageKey): any {
+    const localStorageItems = this.getAllItems();
+    const matchingKeys =
+      Object.keys(localStorageItems).filter(lsi => lsi.indexOf(key) >= 0);
```

Comment:
would it not be more accurate to match from the start, rather than any part of the key?

Reply:
good point, I changed this to use`startsWith` instead